samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.servicebus.samples.queueclientquickstart;
+
+import com.microsoft.azure.servicebus.*;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import org.apache.commons.cli.*;
+import org.apache.log4j.*;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class QueueClientQuickstart {
+    // Connection String for the namespace can be obtained from the Azure portal under the
+    // 'Shared Access policies' section.
+
+    // connection string can be initialized via environment variable
+    private static String connectionString = System.getenv("SB_SAMPLES_CONNECTIONSTRING");
+    // queue name can be initialiyzed via environment variable
+    private static String queueName = System.getenv("SB_SAMPLES_QUEUENAME");
+    // queue client instance
+    private static IQueueClient queueClient;
+    // message send/receive counters
+    private static int totalToSend = 100;
+    private static AtomicInteger totalReceived = new AtomicInteger(0);
+    // log4j logger 
+    private static Logger logger = Logger.getRootLogger();
+
+    public static void main(String[] args) throws Exception {
+        // Parse and evaluate command line. Exit when there's not
+        // enough configuration information in environment and to 
+        // command line to continue
+        if (!parseCommandLine(args)) {
+            return;
+        }
+
+        logger.info("Starting BasicSendReceiveWithQueueClient sample");
+
+        // Create a QueueClient instance using the connection string builder
+        logger.info("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PEEKLOCK);
+
+        // send messages in a loop
+        for (int i = 0; i < totalToSend; i++) {
+            int currentMessageCounter = i;
+            queueClient.sendAsync(new Message("" + i)).thenRunAsync(() -> {
+                logger.info(String.format("Sent message #%d.", currentMessageCounter));
+            });
+        }
```

Comment:
Given that the aim of these samples also include best practices, we should have these inside `try-catch` and have `queueClient.close()` in `finally`

Reply:
This is the Quickstart. Other samples should be more robust, but it's ok for this one to be a bit more optimistic. It's sending messages inside of a loop and that, per-se, probably isn't "best practice". Adding proper handling here means that you put a try/catch with logging and finally inside that loop and then it gets all a bit hard to read for beginners.